app/controllers/carts_controller.rb

```diff
@@ -1,4 +1,14 @@
 class CartsController < ApplicationController
+  def checkout_cart
+    if session[:user_id].nil?
+      flash[:alert] = 'You must login to checkout'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.